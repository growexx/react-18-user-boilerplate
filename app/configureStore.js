import { configureStore } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import { createReduxHistoryContext } from 'redux-first-history';

import { repoApi } from './containers/HomePage/reposApiSlice';
import { appSlice } from './containers/App/slice';
import { apiSlice } from './apiSlice';
import languageSlice from './containers/LanguageProvider/slice';
import loginSlice from './containers/Auth/Login/slice';
import { authApi } from './containers/Auth/Login/authApiSlice';
import twoFactorAuthSlice from './containers/Auth/TwoFactorAuthentication/slice';

const { createReduxHistory, routerMiddleware, routerReducer } =
  createReduxHistoryContext({ history: createBrowserHistory() });

const store = configureStore({
  reducer: {
    router: routerReducer,
    app: appSlice,
    language: languageSlice,
    login: loginSlice,
    twoFactorAuth: twoFactorAuthSlice,
    repos: repoApi.reducer,
    auth: authApi.reducer,
    [apiSlice.reducerPath]: apiSlice.reducer,
  },
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware(),
    apiSlice.middleware,
    routerMiddleware,
  ],
  devTools: process.env.NODE_ENV !== 'production',
});

const history = createReduxHistory(store);

export { store, history };
