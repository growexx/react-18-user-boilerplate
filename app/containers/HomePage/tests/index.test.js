/**
 * Test the HomePage
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';

import { store } from 'configureStore';
import { HomePage } from '../index';

describe('<HomePage />', () => {
  let globalStore;

  beforeAll(() => {
    globalStore = store;
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={globalStore}>
        <IntlProvider locale="en">
          <HomePage loading={false} error={false} repos={[]} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch the repos on mount if a username exists', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={globalStore}>
        <IntlProvider locale="en">
          <HomePage
            username="Not Empty"
            onChangeUsername={() => {}}
            onSubmitForm={submitSpy}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).toHaveBeenCalledTimes(1);
  });

  it('should not call onSubmitForm if username is an empty string', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={globalStore}>
        <IntlProvider locale="en">
          <HomePage onChangeUsername={() => {}} onSubmitForm={submitSpy} />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  it('should not call onSubmitForm if username is null', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={globalStore}>
        <IntlProvider locale="en">
          <HomePage
            username=""
            onChangeUsername={() => {}}
            onSubmitForm={submitSpy}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });
});
