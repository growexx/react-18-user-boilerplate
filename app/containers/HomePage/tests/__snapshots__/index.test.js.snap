// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
.c5 {
  list-style: none;
  margin: 0;
  width: 100%;
  max-height: 30em;
  overflow-y: auto;
  padding: 0 1em;
}

.c4 {
  padding: 0;
  margin: 0;
  width: 100%;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 3px;
  overflow: hidden;
}

.c6 {
  margin: 2em auto;
  width: 40px;
  height: 40px;
  position: relative;
}

.c2 {
  color: black;
  margin-left: 0.4em;
}

.c1 {
  margin-bottom: 1em;
}

.c3 {
  outline: none;
  border: none;
  border-bottom: 1px dotted #999;
  background-color: transparent;
}

.c0 {
  margin: 3em auto;
}

.c0:first-child {
  margin-top: 0;
}

.c7 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
}

.c7:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
}

.c8 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg);
}

.c8:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -1.1s;
  -webkit-animation-delay: -1.1s;
  animation-delay: -1.1s;
}

.c9 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  transform: rotate(60deg);
}

.c9:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -1s;
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}

.c10 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c10:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -0.9s;
  -webkit-animation-delay: -0.9s;
  animation-delay: -0.9s;
}

.c11 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  transform: rotate(120deg);
}

.c11:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -0.8s;
  -webkit-animation-delay: -0.8s;
  animation-delay: -0.8s;
}

.c12 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(150deg);
  -ms-transform: rotate(150deg);
  -webkit-transform: rotate(150deg);
  -ms-transform: rotate(150deg);
  transform: rotate(150deg);
}

.c12:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -0.7s;
  -webkit-animation-delay: -0.7s;
  animation-delay: -0.7s;
}

.c13 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c13:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -0.6s;
  -webkit-animation-delay: -0.6s;
  animation-delay: -0.6s;
}

.c14 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(210deg);
  -ms-transform: rotate(210deg);
  -webkit-transform: rotate(210deg);
  -ms-transform: rotate(210deg);
  transform: rotate(210deg);
}

.c14:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -0.5s;
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

.c15 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(240deg);
  -ms-transform: rotate(240deg);
  -webkit-transform: rotate(240deg);
  -ms-transform: rotate(240deg);
  transform: rotate(240deg);
}

.c15:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -0.4s;
  -webkit-animation-delay: -0.4s;
  animation-delay: -0.4s;
}

.c16 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

.c16:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -0.3s;
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c17 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(300deg);
  -ms-transform: rotate(300deg);
  -webkit-transform: rotate(300deg);
  -ms-transform: rotate(300deg);
  transform: rotate(300deg);
}

.c17:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -0.2s;
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c18 {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  transform: rotate(330deg);
}

.c18:before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  background-color: #999;
  border-radius: 100%;
  -webkit-animation: hRYYZp 1.3s infinite ease-in-out both;
  animation: hRYYZp 1.3s infinite ease-in-out both;
  -webkit-animation-delay: -0.1s;
  -webkit-animation-delay: -0.1s;
  animation-delay: -0.1s;
}

<article>
  <div>
    <section
      class="c0"
    >
      <h2>
        Try me!
      </h2>
      <form
        class="c1"
        role="form"
      >
        <label
          for="username"
        >
          Show Github repositories by
          <span
            class="c2"
          >
            @
          </span>
          <input
            class="c3"
            id="username"
            placeholder="mxstbr"
            type="text"
            value="mxstbr"
          />
        </label>
      </form>
      <div
        class="c4"
      >
        <ul
          class="c5"
        >
          <div
            class="c6"
          >
            <div
              class="c7"
            />
            <div
              class="c8"
            />
            <div
              class="c9"
            />
            <div
              class="c10"
            />
            <div
              class="c11"
            />
            <div
              class="c12"
            />
            <div
              class="c13"
            />
            <div
              class="c14"
            />
            <div
              class="c15"
            />
            <div
              class="c16"
            />
            <div
              class="c17"
            />
            <div
              class="c18"
            />
          </div>
        </ul>
      </div>
    </section>
  </div>
</article>
`;
